src/replica/replica.cpp

```diff
@@ -52,8 +53,12 @@ namespace replication {
 
 const std::string replica::kAppInfo = ".app-info";
 
-replica::replica(
-    replica_stub *stub, gpid gpid, const app_info &app, const char *dir, bool need_restore)
+replica::replica(replica_stub *stub,
+                 gpid gpid,
+                 const app_info &app,
+                 const char *dir,
+                 bool need_restore,
+                 bool need_duplicate)
```

Comment:
Same as `duplicate_if_necessary`, it seems duplication follower and master should also set `need_duplicate` as true.